assets/src/block-editor/SidebarFields/DatePickerSidebarField.js

```diff
@@ -0,0 +1,18 @@
+const {DatePicker, BaseControl} = wp.components;
+
+export const DatePickerSidebarField = ({id, value, setValue, label, forceEndDate = false}) => (
+  <BaseControl id={id} label={label}>
+    <DatePicker
+      id={id}
+      currentDate={value}
+      onChange={date => {
+        if(forceEndDate) {
+          setValue(`${date.slice(0,10)+'T23:59:59'}`);
```

Comment:
I've added this to set the last second of the assigned day as deadline. If no, the Date Picker takes the exact time you're setting. For instance, If my time is 4:54PM, then the deadline is also at the same time.